src/main/java/dan200/computercraft/core/apis/TermAPI.java

```diff
@@ -270,6 +333,65 @@ public static int parseColour( Object[] args, boolean _enableColours ) throws Lu
                 }
                 return null;
             }
+            case 19:
+            case 20:
+            {
+                // setColour/setColor
+                if( !m_environment.isColour() )
```

Comment:
I think we should handle greyscale terminals as follows:
Still allow you set the palette colour on any type of terminal, but just convert the colours to greyscale (by setting each component to (r+g+b)/3) during rendering. This also means we can allow any colour to be set during setTextColour/BackgroundColour